src/js/ProviderManager.js

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/*globals async, DirectoryLinksProvider, Links, PlacesProvider, gUserDatabase, gPinnedLinks, gBlockedLinks*/
+
+/**
+ * Singleton that handles link providers.
+ */
+
+ "use strict";
+
+const ProviderManager = {
+
+  _initialized: false,
+
+  get initialized() {
+    return this._initialized;
+  },
+
+  set initialized(initialized) {
+    this._initialized = initialized;
+  },
+
+  init: async(function*(requestURL) {
```

Comment:
Yes, I think this is a good idea to keep all initialization in one place. I think for now I can add ProviderManager initialization in `initializeNewTabPage()`